packages/react-vapor/src/components/modal/tests/ConfirmationModalProvider.spec.tsx

```diff
@@ -1,155 +1,134 @@
-import {mount, shallow, ShallowWrapper} from 'enzyme';
+import {screen} from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
 import * as React from 'react';
-import {Provider} from 'react-redux';
-
-import {getStoreMock} from '../../../utils/tests/TestUtils';
-import {ModalCompositeConnected} from '../ModalComposite';
+import {renderModal} from 'react-vapor-test-utils';
 import {ConfirmationModalProvider} from '../ConfirmationModalProvider';
+import {ModalCompositeConnected} from '../ModalComposite';
 
 describe('<UnsavedChangeModalProvider/>', () => {
-    let confirmationModalProvider: ShallowWrapper;
-    let heavyConfirmationModalProvider;
-    let regularClickActionSpy: jest.Mock<any, any>;
-    let promptBeforeClickActionSpy: jest.Mock<any, any>;
-    const store = getStoreMock();
-    beforeEach(() => {
-        regularClickActionSpy = jest.fn();
-        promptBeforeClickActionSpy = jest.fn();
-    });
-
-    afterEach(() => {
-        regularClickActionSpy.mockReset();
-        promptBeforeClickActionSpy.mockReset();
-        if (confirmationModalProvider?.exists()) {
-            confirmationModalProvider.unmount();
-        }
-    });
-
-    const shallowMountUnsavedModalProvider = (shouldConfirm: boolean = false) =>
-        shallow(
-            <ConfirmationModalProvider
-                shouldConfirm={shouldConfirm}
-                modalTitle="Are you sure?"
-                modalBodyChildren={<div>some content</div>}
-            >
-                {({promptBefore}) => (
-                    <ModalCompositeConnected
-                        id="👾"
-                        title="🙄"
-                        classes={['mod-slide-in-bottom', 'mod-big', 'mod-stick-bottom']}
-                        modalBodyChildren="🙈 🙉 🙊"
-                        modalFooterChildren={
-                            <button
-                                className="btn"
-                                onClick={() => promptBefore(promptBeforeClickActionSpy) && regularClickActionSpy()}
-                            />
-                        }
-                        modalBodyClasses={['mod-header-padding', 'mod-form-top-bottom-padding']}
-                        docLink={{url: 'https://www.coveo.com', tooltip: {title: 'Go to coveo.com'}}}
-                    />
-                )}
-            </ConfirmationModalProvider>
-        );
-
-    const heavilyMountUnsavedModalProvider = (shouldConfirm: boolean = false) =>
-        mount(
-            <ConfirmationModalProvider
-                shouldConfirm={shouldConfirm}
-                modalTitle="Are you sure?"
-                modalBodyChildren={<div>some content</div>}
-            >
-                {({promptBefore}) => (
-                    <Provider store={store}>
+    describe('when shouldConfirm prop is true:', () => {
+        it('should open the ConfirmationModal to interrupt the leaving action if isDirty is true', () => {
```

Comment:
The `isDirty` has no impact on the prompt being shown or not, it is shown when calling `promptBefore()`.

```suggestion
        it('opens the ConfirmationModal when calling its `promptBefore` prop', () => {
```